src/Cli/func/Actions/HostActions/StartHostAction.cs

```diff
@@ -438,22 +438,25 @@ public override async Task RunAsync()
             var runTask = host.RunAsync();
             var hostService = host.Services.GetRequiredService<WebJobsScriptHostService>();
 
-            await hostService.DelayUntilHostReady();
-
-            var scriptHost = hostService.Services.GetRequiredService<IScriptJobHost>();
-            var httpOptions = hostService.Services.GetRequiredService<IOptions<HttpOptions>>();
-            if (scriptHost != null && scriptHost.Functions.Any())
+            if (hostService.State is not ScriptHostState.Stopping && hostService.State is not ScriptHostState.Stopped)
             {
-                // Checking if in Limelight - it should have a `AzureDevSessionsRemoteHostName` value in local.settings.json.
-                var forwardedHttpUrl = _secretsManager.GetSecrets().FirstOrDefault(
-                    s => s.Key.Equals(Constants.AzureDevSessionsRemoteHostName, StringComparison.OrdinalIgnoreCase)).Value;
-                if (forwardedHttpUrl != null)
+                await hostService.DelayUntilHostReady();
```

Comment:
Do we have a shutdown cancellation token available here? If so, recommend the following:


```suggestion
                await hostService.DelayUntilHostReady().WaitAsync(shutdownToken);
```